packages/ui-components/src/components/PortalProvider/PortalProvider.component.js

```diff
@@ -1,79 +1,142 @@
+        rootRef.current.removeChild(containerRef.current)
+        containerRef.current = null
+      }
+    }
+  }, [rootRef])
 
-Portal.propTypes = {
-  children: PropTypes.any,
+  if (!containerRef?.current) {
+    return null
+  }
+  // return the current of the ref or null if not yet ready
+  return isReady ? containerRef.current : null
 }
 
-Portal.propTypes = {}
-
-/**
- * This provider acts as a container for portals. All portals within a Juno app should be added as children to this.
- * The PortalProvider itself needs to be placed inside the Juno StyleProvider, otherwise styles might not be applied correctly on children of portals.
- *
- * The main task of the PortalProvider is to offer a place (portal) where certain components
- * such as modals are mounted. Many existing libs place such components outside of the
- * current application's DOM tree, because the control over creating and scheduling
- * the components is not with the application but with the lib. This is not a problem
- * as long as the application is in the global document tree. Once shadow root comes
- * into play, it changes. In this case, such components are placed outside of the
- * shadow root and individual app styles are not applied. The PortalProvider solves
- * this problem by creating the portal that lives in the same DOM tree as the actual app.
- *
- * The PortalProvider is appended at the top of the application tree and all lower
- * components are children of it. This means that all children can access the portal.
- * There are two ways you can do this. Via the ProtalProvider.Portal component or via
- * a usePortalRef hook. While the component places all children in the portal, the hook
- * returns a React reference object to the DOM element.
+/** A PortalProvider component that helps using and managing portals.
+ * It renders a portal root container, creates a context to expose a ref the container, a `PortalProvider.Portal` component to render content into a portal, and a `usePortalRef` hook to render content into a portal.
+ * Normally, there is no need to include `PortalProvider` manually, when using `AppShell` `PortalProvider` is already included in the app.
  */
-export const PortalProvider = ({ className = "", id = "", children = null }) => {
-  const ref = useRef()
+export const PortalProvider = ({ children = null, className = "", id = DEFAULT_PORTAL_ROOT_ID }) => {
+  const portalRootRef = useRef()
 
   return (
-    <PortalContext.Provider value={ref}>
+    <PortalContext.Provider value={portalRootRef}>
       {children}
-      <div className={`juno-portal-container ${className}`} id={id} ref={ref} />
+      <div className={`juno-portal-root ${className}`} id={id} ref={portalRootRef} style={portalRootStyles} />
     </PortalContext.Provider>
   )
 }
-// bind Portal to PortalProvider
+
+// Bind Portal to PortalProvider:
 PortalProvider.Portal = Portal
 Portal.displayName = "PortalProvider.Portal"
 
 PortalProvider.propTypes = {
```

Comment:
My suggestion of typescript that propTypes are going to be deprecated by React 19 we have to move on. That's why we need this component on Typescript